doc/reference/files/dune-project/dialect.rst

```diff
@@ -65,3 +65,20 @@ dialect
       This field supports the same sub-fields as ``implementation``.
 
       .. versionchanged:: 3.9 This field is made optional.
+
+
+Default dialects
+----------------
+
+In the default configuration Dune comes with two dialects pre-set and enabled:
+
+* ``ocaml`` for the default OCaml syntax which consumes `.ml` and `.mli` files
+  and uses ``ocamlformat`` for formatting.
+* ``reason`` for the Reason syntax and enabled in `.re`/`.rei` files. For
+  formatting ``refmt`` is used.
+
+When ``(using melange)`` is enabled in the `dune-project` a third dialect is
+enabled:
+
+* ``rescript`` for the ReScript syntax which is used in `.res` and `.resi`
+  files. The formatter in this case is ``rescript_syntax``.
```

Comment:
This should be specified, rescript_syntax only works in Melange v1 and it doesn't support the entire rescript language (as newer versions) in any form in dune

Reply:
As someone who is a bit out of touch with Rescript and Melange, does that mean that `rescript_syntax` can only be used in Melange v1 (and not v2 or v3) and even then only supports an older subset of the language?

I don't know to which extent Melange v1 + ReScript are used but this sounds like a serious restriction and makes me wonder if supporting something of progressively lower relevance as time goes on and Rescript evolves makes sense overall.

Reply:
Exactly, rescript_syntax (https://opam.ocaml.org/packages/rescript-syntax/) was a tool to help people migrate from ReScript to Melange. It's not a feature dune should keep up, since rescript-syntax isn't up-to-date with https://github.com/rescript-lang/rescript-compiler in any form.